wod_predictor/feature_engineering_parts/athlete_info_fe.py

```diff
@@ -15,42 +16,73 @@ def __init__(self, missing_method="zero", **kwargs):
 
     def fit(self, athlete_info_data: pd.DataFrame):
         """
-        Add any initial setup here if needed
+        Add any initial setup here if needed.
         """
         pass
 
     def transform(self, athlete_info_data: pd.DataFrame):
         """
-        Transforms athlete info data
+        Transforms athlete info data:
         - melts data from wide to long format, with each row representing an athlete/year pair
         - converts units to metric
         - creates features from athlete info data
-        - OPTIONAL: fills missing values
+        - calculates and verifies birth year consistency across all reported years
+        - fills missing values if specified
 
         Parameters:
             athlete_info_data (pd.DataFrame): Athlete info data
 
         Returns:
             pd.DataFrame: Transformed athlete info data
         """
+        # Melt data into long format
         athlete_info_melted = self.melt(athlete_info_data)
 
-        # Tconvert data types to numeric
+        # Convert data types to numeric
         athlete_info_numeric = self.fix_units(athlete_info_melted)
 
-        # create features from athlete info data
+        # Calculate birth year based on each reported year and verify consistency
+        athlete_info_numeric['birth_year_calculated'] = (
+            athlete_info_numeric['year'].astype(int) - athlete_info_numeric['age']
+        )
```

Comment:
This makes all birth years strange numbers, if someone is 47 years old in 2022, this value would be -25? Doesn't matter at all in terms of modeling, but might be good to make it interpretable.